### PR TITLE
Fixes #22732 - Add export CSV for RH repos and subs

### PR DESCRIPTION
**Pull request by johnpmitsch**
This adds the export CSV button to both RH repos page
and the Subscription page. It uses the csv functionality
currenlty in foreman with fixed columns for the CSV file.

### CODE REVIEW

File: webpack/scenes/RedHatRepositories/components/SearchBar.js [hunk 1]
```diff
@@ -101,6 +105,12 @@ class SearchBar extends Component {
               this.onSelectFilterType(values);
             }}
           />
+          <Button
+            className="export-csv-button"
+            onClick={() => { api.open('/repositories.csv', repoParams); }}
+          >
+            {'Export CSV'}
```
Comment (waldenraines) [inline webpack/scenes/RedHatRepositories/components/SearchBar.js]: This needs to be translated using `__()`

File: webpack/scenes/RedHatRepositories/index.scss [hunk 2]
```diff
@@ -47,6 +47,10 @@
     margin-bottom: 7px;
   }
 
+  .export-csv-button {
+    margin: 0px 20px;
+  }
+
```
Comment (waldenraines) [inline webpack/scenes/RedHatRepositories/index.scss]: If you wrap the `<button>` in a `<FormGroup>` you won't need this custom css.

**Comment by johnpmitsch** [comment]
> Any idea how well this will work with customizable columns?

Currently, the UI columns won't affect what is included in the CSV file. The CSV columns are hardcoded server side. It would definitely be possible to match the UI columns to CSV columns in later work. In other words, nothing here blocks custom columns in CSV from happening, but changes would have to be made to support this.

I view CSV files as a dump for information that can be easily manipulated, so I don't see the need for custom CSV columns, but that is a different discussion :smile:

> It looks like the export csv button on the RH repos page only exports the enabled repos? Should we change the button to say "Export Enabled as CSV" or something? Or is it worthwhile to look at hooking the button into the enable/available/both selector?

Ya, that is a good idea, its not explicit right now. I can update the button. I'll hold off on adding it for available repos for now, it can be added later if demand for it is there.

**Comment by sjha4** [comment]
Able to implement csv export on other katello pages using changes in the api controller. ACK from me on that front.

**Comment by johnpmitsch** [comment]
When I wrap the button in a form group here:
```
        <Form className="toolbar-pf-actions">
          <FormGroup className="toolbar-pf-filter">
            <Search onSearch={this.onSearch} onSelectSearchList={this.onSelectSearchList} />
          </FormGroup>
  
          <FormGroup className="toolbar-pf-filter">
            <MultiSelect
              value={this.state.filters}
              options={filterOptions}
              onChange={(e) => {
                const values = [...e.target.options]
                  .filter(({ selected }) => selected)
                  .map(({ value }) => value);
                this.onSelectFilterType(values);
              }}
            />
+         </FormGroup>
+         <FormGroup>
_           <Button
              onClick={() => { api.open('/repositories.csv', repoParams); }}
            >
~             {__('Export Enabled as CSV')}
            </Button>
          </FormGroup>
        </Form>
```
this happens:
here

I put the custom class back (but changed the button name), I'm open to trying out more suggestions.

**Comment by johnpmitsch** [comment]
@waldenraines updated according to your comments (with the exception of what I mention above)

**Comment by waldenraines** [comment]
> I put the custom class back (but changed the button name), I'm open to trying out more suggestions.

Here's what I did to get this:

screenshot from 2018-06-04 11-19-30

```diff
diff --git a/webpack/scenes/RedHatRepositories/components/SearchBar.js b/webpack/scenes/RedHatRepositories
index d97abcd..efd0003 100644
--- a/webpack/scenes/RedHatRepositories/components/SearchBar.js
+++ b/webpack/scenes/RedHatRepositories/components/SearchBar.js
@@ -94,19 +94,18 @@ class SearchBar extends Component {
           <Search onSearch={this.onSearch} onSelectSearchList={this.onSelectSearchList} />
         </FormGroup>
 
-        <FormGroup className="toolbar-pf-filter">
-          <MultiSelect
-            value={this.state.filters}
-            options={filterOptions}
-            onChange={(e) => {
-              const values = [...e.target.options]
-                .filter(({ selected }) => selected)
-                .map(({ value }) => value);
-              this.onSelectFilterType(values);
-            }}
-          />
+        <MultiSelect
+          value={this.state.filters}
+          options={filterOptions}
+          onChange={(e) => {
+            const values = [...e.target.options]
+              .filter(({ selected }) => selected)
+              .map(({ value }) => value);
+            this.onSelectFilterType(values);
+          }}
+        />
+        <FormGroup>
           <Button
-            className="export-csv-button"
             onClick={() => { api.open('/repositories.csv', repoParams); }}
           >
             {__('Export Enabled as CSV')}
```

You'll obviously also want to remove the class from the scss.

**Comment by johnpmitsch** [comment]
@waldenraines thanks for that suggestion, I updated. I had to create some custom CSS to keep the gap between the selector and the table, this was orginally done by `toolbar-pf-filter` on the FormGroup that was removed. `toolbar-pf-filter` sets a width of %50, which is why we were getting the broken rows. I created a new class `toolbar-pf-multiselect` to fix this.

**Comment by johnpmitsch** [comment]
@waldenraines ignore what I just said about the custom class, once I updated to the new version of bootstrap selector, it looks ok as is.

**Pull request merged**